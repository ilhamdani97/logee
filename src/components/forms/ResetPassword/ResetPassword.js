import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import Text from '../../fields/Text_V2';
import styles from './styles.scoped.css';
import { Button } from 'logeeport-ui';
import { useSelector } from 'react-redux';

const ResetPassword = (props) => {
  const { handleSubmit, valid } = props;

  const textInputProps = { placeholder: 'Cth. ganendra@gmail.com' };

  const { message, isLoading } = useSelector((s) => s.resetPassword);
  return (
    <>
      <form className={styles.root}>
        <Field
          component={Text}
          inputProps={textInputProps}
          label="Email"
          name="email"
        />
        <Button
          disabled={isLoading || !valid}
          isLoading={isLoading}
          onClick={handleSubmit}
        >
          Kirim
        </Button>
      </form>
      {message && <span className={styles.error}>{message}</span>}
    </>
  );
};

ResetPassword.defaultProps = {
  handleSubmit: () => {},
  valid: true,
};

ResetPassword.propTypes = {
  handleSubmit: PropTypes.func,
  valid: PropTypes.bool,
};

export default ResetPassword;
