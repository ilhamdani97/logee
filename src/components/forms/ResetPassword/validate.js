import { formatEmail } from '@utils/format';

export default function validate(values) {
  const { email } = values;

  return {
    email: !email && 'required' || !formatEmail(email) && 'Format email salah',
  };
}
