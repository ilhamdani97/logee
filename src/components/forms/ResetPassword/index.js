import { reduxForm } from 'redux-form';
import Component from './ResetPassword';
import validate from './validate';

export default reduxForm({
  form: 'resetpassword',
  initialValues: {
    email:'',
  },
  validate,
})(Component);
