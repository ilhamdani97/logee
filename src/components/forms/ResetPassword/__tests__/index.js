import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ResetPassword from '../ResetPassword';
import { useSelector } from 'react-redux';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    isLoading:false,
    message: 'test'
  };
});

describe('src/components/forms/ResetPassword', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ResetPassword />);
    expect(tree).toMatchSnapshot();
    // useSelector.mockImplementationOnce(() => ({ isLoading: true, message: 'error' }));
    // ResetPassword.defaultProps.handleSubmit();
  });
});
