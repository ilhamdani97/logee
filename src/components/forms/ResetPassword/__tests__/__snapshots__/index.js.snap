// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ResetPassword render 1`] = `
<React.Fragment>
  <form>
    <Field
      component={[Function]}
      inputProps={
        Object {
          "placeholder": "Cth. ganendra@gmail.com",
        }
      }
      label="Email"
      name="email"
    />
    <Button
      buttonProps={Object {}}
      className=""
      color="default"
      disabled={false}
      fullwidth={false}
      isLoading={false}
      loadingType="text"
      onClick={[Function]}
      size="medium"
    >
      Kirim
    </Button>
  </form>
  <span>
    test
  </span>
</React.Fragment>
`;
